airflow-core/tests/unit/models/test_deadline.py

```diff
@@ -375,11 +377,25 @@ def test_deadline_database_integration(self, reference, expected_column, session
             if expected_column is not None:
                 result = reference.evaluate_with(session=session, interval=interval, **conditions)
                 mock_fetch.assert_called_once_with(expected_column, session=session, **conditions)
+            elif reference == DeadlineReference.AVERAGE_RUNTIME():
+                with mock.patch("airflow._shared.timezones.timezone.utcnow") as mock_utcnow:
+                    mock_utcnow.return_value = DEFAULT_DATE
+                    with mock.patch.object(session, "execute") as mock_execute:
+                        # Mock the result object that execute() returns
+                        mock_result = mock.Mock()
+                        mock_scalars = mock.Mock()
+                        mock_scalars.all.return_value = [3600]
+                        mock_result.scalars.return_value = mock_scalars
+                        mock_execute.return_value = mock_result
+                        result = reference.evaluate_with(session=session, interval=interval, dag_id=DAG_ID)
+                        mock_fetch.assert_not_called()
+                        # Should be DEFAULT_DATE + default average deadline of 48 hours + the 1 hour mocked execution time
```

Comment:
This is testing the path of the default, but where is the test for the path where we have enough history to compute an average?